Subject: [PATCH] hw02 - добавил страницы: Главная, Профиль, Авторизация, Регистрация
---
Index: resources/js/router/index.js
===================================================================
diff --git a/resources/js/router/index.js b/resources/js/router/index.js
deleted file mode 100644
--- a/resources/js/router/index.js	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
+++ /dev/null	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
@@ -1,19 +0,0 @@
-import { createRouter, createWebHistory } from 'vue-router';
-import Home from '../components/Home.vue';
-import About from '../components/About.vue';
-import Login from '../components/Login.vue';
-import Profile from '../components/Profile.vue';
-import Register from '../components/Register.vue';
-
-const routes = [
-    {path: '/', component: Home},
-    {path: '/about', component: About},
-    {path: '/login', component: Login},
-    {path: '/profile', component: Profile},
-    {path: '/register', component: Register},
-];
-
-export default createRouter({
-    history: createWebHistory(),
-    routes,
-});
Index: resources/js/app.js
===================================================================
diff --git a/resources/js/app.js b/resources/js/app.js
deleted file mode 100644
--- a/resources/js/app.js	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
+++ /dev/null	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
@@ -1,6 +0,0 @@
-import './bootstrap';
-import { createApp } from 'vue';
-import router from './router';
-import App from './App.vue';
-
-createApp(App).use(router).mount('#app');
Index: resources/js/App.vue
===================================================================
diff --git a/resources/js/App.vue b/resources/js/App.vue
deleted file mode 100644
--- a/resources/js/App.vue	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
+++ /dev/null	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
@@ -1,18 +0,0 @@
-<template>
-    <div>
-        <nav class="p-4 border-b flex gap-4">
-            <router-link to="/">Главная</router-link>
-            <router-link to="/login">Вход</router-link>
-            <router-link to="/register">Регистрация</router-link>
-            <router-link to="/profile">Профиль</router-link>
-            <router-link to="/about">О нас</router-link>
-        </nav>
-        <main class="p-4">
-            <router-view />
-        </main>
-    </div>
-</template>
-
-<script>
-export default {}
-</script>
Index: resources/views/app.blade.php
===================================================================
diff --git a/resources/views/app.blade.php b/resources/views/app.blade.php
deleted file mode 100644
--- a/resources/views/app.blade.php	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
+++ /dev/null	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
@@ -1,12 +0,0 @@
-<!DOCTYPE html>
-<html lang="ru">
-<head>
-    <meta charset="UTF-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>EventPass</title>
-    @vite(['resources/css/app.css', 'resources/js/app.js'])
-</head>
-<body>
-<div id="app"></div>
-</body>
-</html>
Index: resources/js/components/Profile.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/resources/js/components/Profile.vue b/resources/frontend/src/components/Profile.vue
rename from resources/js/components/Profile.vue
rename to resources/frontend/src/components/Profile.vue
--- a/resources/js/components/Profile.vue	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
+++ b/resources/frontend/src/components/Profile.vue	(date 1763470098642)
@@ -9,6 +9,8 @@
 </template>
 
 <script>
+import { API_URL } from '../config.js';
+
 export default {
     data() {
         return {
@@ -17,7 +19,7 @@
     },
     async mounted() {
         // Заглушка - в реальном приложении брать ID из авторизации
-        const response = await fetch('/api/user/1');
+        const response = await fetch(API_URL + '/api/user/1');
         const data = await response.json();
         this.user = data.user;
     }
Index: resources/frontend/src/config.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/resources/frontend/src/config.js b/resources/frontend/src/config.js
new file mode 100644
--- /dev/null	(date 1763469993759)
+++ b/resources/frontend/src/config.js	(date 1763469993759)
@@ -0,0 +1,2 @@
+
+export const API_URL = import.meta.env.VITE_API_URL;
Index: resources/js/components/Home.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/resources/js/components/Home.vue b/resources/frontend/src/components/Home.vue
rename from resources/js/components/Home.vue
rename to resources/frontend/src/components/Home.vue
--- a/resources/js/components/Home.vue	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
+++ b/resources/frontend/src/components/Home.vue	(date 1763470054201)
@@ -10,6 +10,7 @@
 </template>
 
 <script>
+import { API_URL } from '../config';
 export default {
     data() {
         return {
@@ -17,7 +18,7 @@
         }
     },
     async mounted() {
-        const response = await fetch('/api/events');
+        const response = await fetch(API_URL + '/api/events');
         const data = await response.json();
         this.events = data.events;
     }
Index: compose.yaml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/compose.yaml b/compose.yaml
--- a/compose.yaml	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
+++ b/compose.yaml	(date 1763468779143)
@@ -43,6 +43,19 @@
         networks:
             - sail
 
+    frontend:
+        image: node:20-alpine
+        working_dir: /app
+        volumes:
+            - ./resources/frontend:/app
+        command: sh -c "npm install && npm run dev -- --host 0.0.0.0 --port 3000"
+        ports:
+            - "3000:3000"
+        networks:
+            - sail
+        depends_on:
+            - php
+
     mysql:
         image: 'mysql/mysql-server:8.0'
         ports:
Index: routes/web.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/routes/web.php b/routes/web.php
--- a/routes/web.php	(revision 6e137a7b0b23fb89224d75cf25c24b93dcc0fba2)
+++ b/routes/web.php	(date 1763535478252)
@@ -2,6 +2,6 @@
 
 use Illuminate\Support\Facades\Route;
 
-Route::get('/{any}', function () {
-    return view('app');
-})->where('any', '.*');
+Route::get('/', function() {
+    return view('welcome');
+});
diff --git a/resources/js/components/Register.vue b/resources/frontend/src/components/Register.vue
rename from resources/js/components/Register.vue
rename to resources/frontend/src/components/Register.vue
diff --git a/resources/js/components/Login.vue b/resources/frontend/src/components/Login.vue
rename from resources/js/components/Login.vue
rename to resources/frontend/src/components/Login.vue
diff --git a/resources/js/components/About.vue b/resources/frontend/src/components/About.vue
rename from resources/js/components/About.vue
rename to resources/frontend/src/components/About.vue
